internal/signature/signature.go

```diff
@@ -0,0 +1,110 @@
+package signature
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+)
+
+// FIXME FIXME: MIME type? Int? String?
+// An interface with a name, parse methods?
+type FormatID string
+
+const (
+	SimpleSigningFormat FormatID = "simple-signing"
+	CosignFormat        FormatID = "cosign-json"
+	// Update also FormatName below
+)
+
+// FormatName returns a human-readable format name for format
+func FormatName(formatID FormatID) string {
+	switch formatID {
+	case SimpleSigningFormat, CosignFormat:
+		return string(formatID)
+	default:
+		return fmt.Sprintf("Unrecognized format %q", string(formatID))
```

Comment:
Thanks, that’s a good point.

OTOH every current user is something like
```go
   return fmt.Errorf("Unsupported signature format %s", signature.FormatName(newSigWithFormat.FormatID()))
```
where returning an error would just be annoying.

So we probably want `signature.UnsupportedFormatError(sig)` that just does all of that in a single call, instead of `FormatName`. And we can add `FormatName` later, after adding a user for it (some kind of “signature inspect” functionality?).

Reply:
Yes, that sounds like a nice approach :+1:

Reply:
Updated now.